R/workflows.R

```diff
@@ -0,0 +1,26 @@
+workflow_validate <- function(path, tasks) {
+  reports <- orderly::orderly_list(root = path)
+  missing_deps <- lapply(names(tasks), function(task) {
+    if (!(task %in% reports)) {
+      stop(sprintf("Task with name '%s' cannot be found.", task))
+    }
+    get_missing_dependencies(task, path, tasks)
+  })
+  list(missing_dependencies = setNames(missing_deps, names(tasks)))
+}
+
+get_missing_dependencies <- function(task, path, tasks) {
+  ## Not ideal that we build a graph of all dependencies then throw all
+  ## these away deeper than 1 level, should probably add support for that
+  ## in orderly to avoid unnecessary computation
+  graph <- orderly::orderly_graph(task, root = path, direction = "upstream",
+                                  use = "src")
+  dependencies <- lapply(graph$root$children, function(vertex) {
+    scalar(vertex$name)
+  })
+  ## TODO: consider search query dependencies i.e.
```

Comment:
At the moment the graph only returns `latest` dependencies, we want to also depend on if the yml contains something like
```
parameters:
  nmin: ~
depends:
  preprocess:
    id: latest(nmin == nmin)
```

Reply:
I added a test of this in orderly, orderly will return for `latest` dependencies and `latest` with some search parameter so it already does what we want. See https://github.com/vimc/orderly/pull/278/files#r612514715